### PR TITLE
Travis CI configuration

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

**Comment by yamt** [comment]
This would have caught a recent bug: https://github.com/apache/incubator-nuttx/pull/181

**Comment by yamt** [comment]
This patch is the step 4, 5 of https://docs.travis-ci.com/user/tutorial/#to-get-started-with-travis-ci
Someone with permission on this repo can do the step 1, 2, 3.

**Comment by xiaoxiang781216** [comment]
> This patch is the step 4, 5 of https://docs.travis-ci.com/user/tutorial/#to-get-started-with-travis-ci
> Someone with permission on this repo can do the step 1, 2, 3.

We have to use Jenkins and server from Apache to conform Apache requirement:
https://github.com/apache/incubator-nuttx/pull/174
If you are interesting in setup the automation build and test, @liuguo09, @maht and you can work together.

**Comment by yamt** [comment]
even if you don't activate it on this repo, this is convenient for people working on their fork on github.

**Comment by davids5** [comment]
What is the harm in merging this to see how well it works. Is it not completely over low-risk?

File: boards/sim/sim/sim/configs/ostest/defconfig [hunk 1]
```diff
@@ -10,10 +10,12 @@ CONFIG_ARCH_BOARD="sim"
 CONFIG_ARCH_BOARD_SIM=y
```
Comment (patacongo) [inline boards/sim/sim/sim/configs/ostest/defconfig]: This change seems out of place with the other commits in the PR.  Perhaps it should be a separate PR?

File: tools/travis/travis.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#! /bin/sh
```
Comment (patacongo) [inline tools/travis/travis.sh]: PLEASE FIX:  All .sh files reequire (currently) a BSD 3-clause license.  No unlicensed shell scripts may come in.

(License will, eventually, be changed to Apache).

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+sudo: required
```
Comment (patacongo) [inline .travis.yml]: Question:  Do .yml files require licenses and headers?  I would think the answer is YES. ??

**Comment by justinmclean** [comment]
ASF policy on headers is reasonable clear [1], but if in doubt add a header. BSD headers should not be added to new files ASF ALv2 headers should be used.

1. https://www.apache.org/legal/src-headers.html#headers

File: boards/sim/sim/sim/configs/ostest/defconfig (hunk 1 above)
Comment (yamt) [reply boards/sim/sim/sim/configs/ostest/defconfig]: this change is required by other changes as otherwise ./nuttx just hangs after running the tests.

**Comment by yamt** [comment]
i added apache license to shell scripts

File: .travis.yml (hunk 3 above)
Comment (yamt) [reply .travis.yml]: i added apache license to this file too as this yaml file contains shell script snippets.

File: tools/travis/travis.sh (hunk 2 above)
Comment (yamt) [reply tools/travis/travis.sh]: i added apache license

**Comment by yamt** [comment]
any takers? eventually i want to add windows build which i can't test locally. i guess it's possible as travis ci provides windows env.

**Comment by maht** [comment]
I think is great to have this code available, but maybe it should be placed in `incubator-nuttx-testing`, as requested by @patacongo [1] and others.

Maybe this preference of a separate repo for the CI code is not settled on stone and it can be reconsidered, but other people then will require to put their own scripts on main repo.

Can the CircleCI script be rewritten to clone the testing repo and move the logic there? I know maybe is a PITA to move it, but maybe would be easier to be accepted, and that is what I have been doing for other CI scripts, like the Jenkins one I have under review in that repo: put a "placeholder" Jenkinsfile in main repo so the CI can "hook" to it and the actual logic in the testing one.

[1] https://www.mail-archive.com/dev@nuttx.apache.org/msg00436.html

**Comment by xiaoxiang781216** [comment]
Hi @yamt , @liuguo09 will look this patch and try to integrate with cibuild.sh.

**Comment by xiaoxiang781216** [comment]
Let's track CI issue by:
https://github.com/apache/incubator-nuttx/pull/261
